### ISSUE TITLE
MD051: Add an option to ignore case for automatically-generated heading identifiers

### ISSUE THREAD

**Issue by Stardidi** [DavidAnson/markdownlint#1274]
The example from the [documentation](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md051.md) reports `MD051/link-fragments: Link fragments should be valid [Expected #heading-name; Actual: #Heading-Name]`

```markdown
# Heading Name

[Link](#Heading-Name)
```

I expect the behavior as described in the example. The actual logic appears to have a bug.

**Comment by DavidAnson** [comment]
The documentation you link explains why the example you show is reported as a violation:

> Link fragments may be handled case-sensitively, so this rule requires fragments to exactly match the GitHub heading algorithm.

**Comment by DerGary** [comment]
Why is this example wrong? It exactly matches the case, so when the link anchor is treated case-sensitive than an exactly matching case should be valid and no violation! Is there a way to change the behaviour of this rule to ignore case or match case exactly??

**Comment by DerGary** [comment]
I got that it implements the GitHub algorithm but the wording in the documentation that `Link fragments may be handled case-sensitively` is misleading. For me this reads like this `because link fragments can be handled either case sensitive or case insensitive we implemented the GitHub algorithm which handles everyting case sensitive`

**Comment by Stardidi** [comment]
The parsers we use are also case-sensitive, so that enhancement would help. But they also handle whitespace differently, so maybe not sufficiently... 

However I had indeed read this documentation wrong, I read it as `different markdown implementations handle case sensitivity differently, so we support both`, this was reinforced by may of the other examples being _correct_  markdown and this one being _incorrect_. Maybe some ✅ and ❌ in the docs could help accentuate when something is valid/invalid?
